frontend/packages/console-shared/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { DraggableData, DraggableCore } from 'react-draggable';
+import { useSize } from './drawer-utils';
+import './Drawer.scss';
+
+type DrawerProps = {
+  height?: number;
+  defaultHeight?: number;
+  open?: boolean;
+  defaultOpen?: boolean;
+  maxHeight?: number | string;
+  isResizable?: boolean;
+  header: React.ReactNode;
+  headerClassName?: string;
+  onChange?: (open: boolean, height: number) => void;
+};
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  defaultHeight,
```

Comment:
Probably good to have some default height if none supplied.
```suggestion
  defaultHeight = 300,
```